src/component/panels/ZonesPanel/ZonesPanel.tsx

```diff
@@ -154,9 +170,15 @@ function ZonesPanelInner({ zones, activeTab, xDomain, yDomain, experiment }) {
     setFlipStatus(false);
   }, []);
 
-  const visibilityHandler = useCallback((key) => {
-    Events.emit('onZonesVisibilityChange', { key });
-  }, []);
+  const handleSetShowZones = useCallback(() => {
```

Comment:
using useCallback or useMemo for things that are passed to a component that is not memoized is actually a (very small) decrease in performance

In general our approach to performance is do not worry about it too much at first unless there are heavy computations involved, or you are working with state (typically in a react context) which trigger the re-rendering of the whole page.

At some point we will refocus on performance looking at bottlenecks, but it is not a priority today.

Reply:
Do you want me to revert states changes ?

Reply:
No, just remove the 3 next `useCallback` as you can just declare regular functions without hurting performance

Reply:
I invite you to read
https://www.joshwcomeau.com/react/why-react-re-renders/
https://www.joshwcomeau.com/react/usememo-and-usecallback/

Reply:
> just remove the 3 next `useCallback`

Done,fixed


> I invite you to read

Thank you